ccl/changefeedccl: TestChangefeedSchemaChangeBackfillCheckpoint failed

ccl/changefeedccl.TestChangefeedSchemaChangeBackfillCheckpoint [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/11169740?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/11169740?buildTab=artifacts#/) on master @ [4fe2a80d81c6fc5a3da3c7c44c5fc38da67e0367](https://github.com/cockroachdb/cockroach/commits/4fe2a80d81c6fc5a3da3c7c44c5fc38da67e0367):


```
=== RUN   TestChangefeedSchemaChangeBackfillCheckpoint
    test_log_scope.go:167: test logs captured to: /artifacts/tmp/_tmp/905465f697dc04c39f7d8a82cb057bfe/logTestChangefeedSchemaChangeBackfillCheckpoint1752657638
    test_log_scope.go:81: use -show-logs to present logs inline
    changefeed_test.go:1870: random seed: -7347133967698241246
=== CONT  TestChangefeedSchemaChangeBackfillCheckpoint
    changefeed_test.go:2115: -- test log scope end --
test logs left over in: /artifacts/tmp/_tmp/905465f697dc04c39f7d8a82cb057bfe/logTestChangefeedSchemaChangeBackfillCheckpoint1752657638
--- FAIL: TestChangefeedSchemaChangeBackfillCheckpoint (9.54s)
=== RUN   TestChangefeedSchemaChangeBackfillCheckpoint/cloudstorage
    helpers_test.go:862: making server as secondary tenant
    helpers_test.go:943: making cloudstorage feed factory
    changefeed_test.go:2100: 
        	Error Trace:	github.com/cockroachdb/cockroach/pkg/ccl/changefeedccl/changefeed_test.go:2100
        	            				github.com/cockroachdb/cockroach/pkg/ccl/changefeedccl/helpers_test.go:1134
        	Error:      	Should be false
        	Test:       	TestChangefeedSchemaChangeBackfillCheckpoint/cloudstorage
        	Messages:   	span should not have been resolved: /Tenant/10/Table/106/{2-3}
    --- FAIL: TestChangefeedSchemaChangeBackfillCheckpoint/cloudstorage (9.52s)
```
<p>Parameters: <code>TAGS=bazel,gss</code>
, <code>stress=true</code>
</p>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
/cc @cockroachdb/cdc
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestChangefeedSchemaChangeBackfillCheckpoint.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>


Jira issue: CRDB-30307

--- a/pkg/ccl/changefeedccl/changefeed_test.go
+++ b/pkg/ccl/changefeedccl/changefeed_test.go
@@ -2092,9 +2092,15 @@ func TestChangefeedSchemaChangeBackfillCheckpoint(t *testing.T) {
 		// Pause job to avoid race on the resolved array
 		require.NoError(t, jobFeed.Pause())
 
+		// NB: With the declarative schema changer, there is a primary index swap,
+		// so the primary index span will change.
+		freshFooDesc := desctestutils.TestingGetPublicTableDescriptor(
+			s.SystemServer.DB(), s.Codec, "d", "foo")
+		tableSpanAfter := freshFooDesc.PrimaryIndexSpan(s.Codec)
+
 		// Verify that none of the resolved spans after resume were checkpointed.
 		for _, sp := range resolved {
-			require.Falsef(t, !sp.Equal(tableSpan) && secondCheckpoint.Contains(sp.Key), "span should not have been resolved: %s", sp)
+			require.Falsef(t, !sp.Equal(tableSpanAfter) && secondCheckpoint.Contains(sp.Key), "span should not have been resolved: %s", sp)
 		}
 	}
 

